backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java

```diff
@@ -182,6 +186,53 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
                 .build();
     }
 
+    private NodeTool createNodeToolWithDirection(NodeDescription nodeDescription, EClass eClass, NodeContainmentKind nodeKind, FeatureDirectionKind direction) {
+        var builder = this.diagramBuilderHelper.newNodeTool();
+
+        var setDirection = this.viewBuilderHelper.newSetValue()
+                .featureName("direction")
+                .valueExpression(direction.getLiteral());
+
+        var callElementInitializerService = this.viewBuilderHelper.newChangeContext()
+                .expression("aql:self.elementInitializer()")
+                .children(setDirection.build());
+
+        var changeContextNewInstance = this.viewBuilderHelper.newChangeContext()
+                .expression("aql:newInstance")
+                .children(callElementInitializerService.build());
+
+        var createEClassInstance = this.viewBuilderHelper.newCreateInstance()
+                .typeName(SysMLMetamodelHelper.buildQualifiedName(eClass))
+                .referenceName(SysmlPackage.eINSTANCE.getRelationship_OwnedRelatedElement().getName())
+                .variableName("newInstance")
+                .children(changeContextNewInstance.build());
+
+        var createView = this.diagramBuilderHelper.newCreateView()
+                .containmentKind(nodeKind)
+                .elementDescription(nodeDescription)
+                .parentViewExpression("aql:selectedNode")
+                .semanticElementExpression("aql:newInstance")
+                .variableName("newInstanceView");
+
+        var changeContexMembership = this.viewBuilderHelper.newChangeContext()
+                .expression("aql:newFeatureMembership")
+                .children(createEClassInstance.build(), createView.build());
+
+        var createMembership = this.viewBuilderHelper.newCreateInstance()
+                .typeName(SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getFeatureMembership()))
+                .referenceName(SysmlPackage.eINSTANCE.getElement_OwnedRelationship().getName())
+                .variableName("newFeatureMembership")
+                .children(changeContexMembership.build());
+
+        var directionLabel = direction.getLiteral().substring(0, 1).toUpperCase() + direction.getLiteral().substring(1);
```

Comment:
You're allowed to use `org.apache.commons.lang3.StringUtils#capitalize`

Reply:
Done